lib/internal/modules/cjs/helpers.js

```diff
@@ -32,11 +32,14 @@ let debug = require('internal/util/debuglog').debuglog('module', (fn) => {
 // TODO: Use this set when resolving pkg#exports conditions in loader.js.
 const cjsConditions = new SafeSet(['require', 'node', ...userConditions]);
 
-function loadNativeModule(filename, request) {
+function loadNativeModule(filename, request, skipCompile = false) {
```

Comment:
Shouldn't this be called after the opposite of `skipCompile`? Eg `publicLoader` or `publicModule` or `doCompile` or something like that?

Reply:
It should cause `loadNativeModule()` to skip calling `mod.compileForPublicLoader()` when it’s truthy.

Reply:
Ah ok makes sense.